tests/dracut_tests/test_driver_updates.py

```diff
@@ -433,10 +434,11 @@ class LoadDriversTestCase(unittest.TestCase):
     def test_basic(self, call):
         """load_drivers: runs depmod and modprobes all named modules"""
         modnames = ['mod1', 'mod2']
-        load_drivers({name: [name] for name in modnames})
+        moddict = collections.OrderedDict({name: [name] for name in modnames})
```

Comment:
Ordered dict doesn't work this way. Now you create a normal dict (which is not ordered) and then add it to ``OrderedDict`` so in the end it won't be ordered at all. The normal dict randomize the order before it gets to ``OrderedDict``.

You need to add items to this dict one by one:
```
for name in modnames:
    ordered_dict[name] = name
```